sky/packages/sky/lib/src/widgets/snack_bar.dart

```diff
@@ -38,7 +38,7 @@ class SnackBarAction extends Component {
   }
 }
 
-class SnackBar extends Component {
+class SnackBar extends StatefulComponent {
```

Comment:
I've tried the new AnimatedComponent route. I think the resulting code is pretty reasonable. The only boilerplate you have now is the syncConstructorArguments().